File: TP/TP2/Code/4AB.cpp
/* Afficher le triangle de Pascal */
#include "sunlib.hpp"
#include <iostream>

/* Procédure */

void trianglePascal(int n)
{
  for (int i = 0; i <= n; ++i) {
    for (int j = 0; j < i + 1; ++j) {
      std::cout << combin(i, j) << " " << std::flush;
    }
    std::cout << std::endl;
  }
}

/* Utilisation */

int main()
{
  trianglePascal(4);
  
  return 0;
}

File: TP/TP2/Code/Python/4AB.py
#!/usr/bin/python
# Afficher le triangle de Pascal
from math import factorial as fact


def combin(n, p):
    return int(fact(n) / (fact(p) * fact(n - p)))


def triangele_pascal(val):
    for i in range(val + 1):
        for j in range(i + 1):
            print(combin(i, j), '', end='')
        print('')


triangele_pascal(4)  # Debug
